mmcls/core/hook/NumClassCheckHook.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) OpenMMLab. All rights reserved
+from mmcv.runner.hooks import Hook
+
+from .builder import HOOKS
```

Comment:
A problem is that the hooks are commonly built in mmcv, but mmcv cannot get the registry of mmcls.
It's a design mistake and we will fix it in the next generation design. But by now, the better solution is to change the name of the hook and still register it to the registry of mmcv. Maybe `ClassNumCheckHook`?
It's not an elegant solution, but works.